### PR TITLE
Build connectivity layer to polygon API

### PR DESCRIPTION
**Pull request by Cebix90**
Builded connectivity layer to polygon API with tests

### CODE REVIEW

File: src/main/java/com/cebix/investmenttrackerapp/handlers/MarketStatusAPIHandler.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.util.UriComponentsBuilder;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+
+@Component
+public class MarketStatusAPIHandler {
+    private final String MARKET_STATUS_URL = "https://api.polygon.io/v1/marketstatus/now";
```
Comment (blazejprzyluski) [inline src/main/java/com/cebix/investmenttrackerapp/handlers/MarketStatusAPIHandler.java:15]: only static components should be in capital letters, please make this one static

File: src/main/java/com/cebix/investmenttrackerapp/handlers/MarketStatusAPIHandler.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.util.UriComponentsBuilder;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+
+@Component
```
Comment (blazejprzyluski) [inline src/main/java/com/cebix/investmenttrackerapp/handlers/MarketStatusAPIHandler.java:13]: Are you autowiring this somewhere?

File: src/main/java/com/cebix/investmenttrackerapp/handlers/StocksAPIHandler.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.util.UriComponentsBuilder;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+import java.time.LocalDate;
+
+@Component
+public class StocksAPIHandler {
+    private final String STOCKS_URL = "https://api.polygon.io/v2/aggs/ticker/";
```
Comment (blazejprzyluski) [inline src/main/java/com/cebix/investmenttrackerapp/handlers/StocksAPIHandler.java:16]: only static components should be in capital letters, please make this one static

File: src/main/java/com/cebix/investmenttrackerapp/handlers/StocksAPIHandler.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        LocalDate toDate = LocalDate.parse(to);
+
+        if (fromDate.isAfter(toDate)) {
+            return Mono.error(new RuntimeException("The parameter 'to' cannot be a time that occurs before 'from'"));
+        }
+
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(STOCKS_URL + "{ticker}/range/{multiplier}/{timespan}/{from}/{to}")
+                .queryParam("adjusted", true)
+                .queryParam("sort", "asc")
+                .queryParam("limit", 120)
```
Comment (blazejprzyluski) [inline src/main/java/com/cebix/investmenttrackerapp/handlers/StocksAPIHandler.java:40]: Maybe this param could be configurable?

File: src/main/java/com/cebix/investmenttrackerapp/handlers/TechnicalIndicatorsAPIHandler.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class TechnicalIndicatorsAPIHandler {
+    private final String TECHNICAL_INDICATORS_URL = "https://api.polygon.io/v1/indicators/";
```
Comment (blazejprzyluski) [inline src/main/java/com/cebix/investmenttrackerapp/handlers/TechnicalIndicatorsAPIHandler.java:18]: static

File: src/main/java/com/cebix/investmenttrackerapp/handlers/StocksAPIHandler.java (hunk 4 above)
Comment (Cebix90) [reply src/main/java/com/cebix/investmenttrackerapp/handlers/StocksAPIHandler.java:40]: added limit parameter to method

File: src/main/java/com/cebix/investmenttrackerapp/handlers/MarketStatusAPIHandler.java (hunk 2 above)
Comment (Cebix90) [reply src/main/java/com/cebix/investmenttrackerapp/handlers/MarketStatusAPIHandler.java:13]: no, changed implementation for webclient variable

**Review by BlazejPrzyl — APPROVED** [review]
Approved!

**Pull request merged**